File: c/space-age/src/space_age.c
#include "space_age.h"
#include <math.h>

float age(planet_t planet, int64_t seconds) {
    switch(planet) {
        case MERCURY: return seconds / 7600543.8; break;
        case VENUS: return seconds / 19414149.1; break;
        case EARTH: return seconds / 31557600.0; break;
        case MARS: return seconds / 59354032.7; break;
        case JUPITER: return seconds / 374355659.1; break;
        case SATURN: return seconds / 929292362.9; break;
        case URANUS: return seconds / 2651370019.3; break;
        case NEPTUNE: return seconds / 5200418560.0; break;
    };

    return NAN;
}

File: go/space-age/space_age.go
package spaceage

type Planet string

func Age(seconds float64, planet Planet) float64 {
	switch planet {
	case "Mercury":
		return seconds / 7600543.8
	case "Venus":
		return seconds / 19414149.1
	case "Earth":
		return seconds / 31557600.0
	case "Mars":
		return seconds / 59354032.7
	case "Jupiter":
		return seconds / 374355659.1
	case "Saturn":
		return seconds / 929292362.9
	case "Uranus":
		return seconds / 2651370019.3
	case "Neptune":
		return seconds / 5200418560.0
	default:
		return -1
	}
}
